src/core/gui/XournalView.cpp

```diff
@@ -45,7 +45,7 @@ XournalView::XournalView(GtkWidget* parent, Control* control, ScrollHandling* sc
     InputContext* inputContext = new InputContext(this, scrollHandling);
     this->widget = gtk_xournal_new(this, inputContext);
     // we need to refer widget here, because we unref it somewhere twice!?
-    g_object_ref(this->widget);
+    // g_object_ref(this->widget);
```

Comment:
I'd say we need to call `g_object_ref_sink` here, and a `g_object_unref` in `~XournalView`.

Did we ever consider using gtkmm (and cairomm, and popplermm, and so on) to stop worrying about those ref count headaches?

Reply:
> I'd say we need to call g_object_ref_sink here, and a g_object_unref in ~XournalView.

Not sure if references from g_object_new are floating. But I'll read the documentation again.
To use g_object_unref would be the correct way, but since something is still holding a reference the deletion is delayed. Control is already deleted then and we have the same mess. So to fix that we have to dig way deeper into the cleanup procedure, or live with that for know.

About gtkmm we thought about that and had a huge debate @lehmanju made an attempt to rewrite the GUI with gtkmm4.
So in principle there is only the fact against it, that it's a lot of work.

Reply:
GTK widget constructors return floating references.

How about just banning g_object_ref/unref/... in the wild and use a unique_ptr-like construct for owning references? (btw this is not the only place where ref management is a bit odd)

Reply:
Well, wrapping every ref/unref in RAII structures would still be quite a bit of work (either by using glibmm's Glib::RefPtr, or by implementing our own).

Reply:
Right, just read it again, I will change that. 

To get rid of the `gtk_widget_destroy` function, the widget also needs to be removed from the parent container.

@danemtsov I already created a similar construct:
https://github.com/xournalpp/xournalpp/blob/fb6254599268dbb71ebfa51d415e64a3f328777c/src/util/include/util/PathUtil.h#L24

But for gtkwidgets we must rewrite an own smart ptr. Something like this: https://www.boost.org/doc/libs/1_78_0/libs/smart_ptr/doc/html/smart_ptr.html#intrusive_ptr

Reply:
`Glib::RefPtr` is only a alias to `std::shared_ptr` (gtkmm4). So this wont even work we would have to replace everything with the corresponding gtkmm type (also we have to rewrite the xournal widget or wrap it).